src/pages/workspace/workflows/WorkspaceWorkflowsApproverPage.tsx

```diff
@@ -0,0 +1,201 @@
+            }
+
+            const details = personalDetails?.[accountID];
+            if (!details) {
+                Log.hmmm(`[WorkspaceMembersPage] no personal details found for policy member with accountID: ${accountID}`);
+                return;
+            }
+
+            const isOwner = policy?.owner === details.login;
+            const isAdmin = policyMember.role === CONST.POLICY.ROLE.ADMIN;
+
+            let roleBadge = null;
+            if (isOwner || isAdmin) {
+                roleBadge = (
+                    <Badge
+                        text={isOwner ? translate('common.owner') : translate('common.admin')}
+                        textStyles={styles.textStrong}
+                        badgeStyles={[styles.justifyContentCenter, StyleUtils.getMinimumWidth(60), styles.badgeBordered]}
+                    />
+                );
+            }
+
+            const formattedMember = {
+                keyForList: accountIDKey,
+                accountID,
+                isSelected: policy?.approver === details.login,
+                isDisabled: policyMember.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE || !isEmptyObject(policyMember.errors),
+                text: formatPhoneNumber(PersonalDetailsUtils.getDisplayNameOrDefault(details)),
+                alternateText: formatPhoneNumber(details?.login ?? ''),
+                rightElement: roleBadge,
+                icons: [
+                    {
+                        source: UserUtils.getAvatar(details.avatar, accountID),
+                        name: formatPhoneNumber(details?.login ?? ''),
+                        type: CONST.ICON_TYPE_AVATAR,
+                        id: accountID,
+                    },
+                ],
+                errors: policyMember.errors,
+                pendingAction: policyMember.pendingAction,
+            };
+
+            if (policy?.approver === details.login) {
+                approverDetails.push(formattedMember);
+            } else {
+                policyMemberDetails.push(formattedMember);
+            }
+        });
+        return [policyMemberDetails, approverDetails];
+    }, [personalDetails, policyMembers, translate, policy?.approver, StyleUtils, isDeletedPolicyMember, policy?.owner, styles]);
+
+    const sections: MembersSection[] = useMemo(() => {
+        const sectionsArray: MembersSection[] = [];
+
+        if (searchTerm !== '') {
+            const filteredOptions = [...formattedApprover, ...formattedPolicyMembers].filter((option) => {
+                const parsedPhoneNumber = parsePhoneNumber(LoginUtils.appendCountryCode(Str.removeSMSDomain(searchTerm)));
+                const searchValue = parsedPhoneNumber.possible ? parsedPhoneNumber.number?.e164 ?? '' : searchTerm.toLowerCase();
+
+                const isPartOfSearchTerm = !!option.text?.toLowerCase().includes(searchValue) || !!option.login?.toLowerCase().includes(searchValue);
+                return isPartOfSearchTerm;
```

Comment:
I feel like this is being used at a lot of places. Is there a way we can make this a utility function?

Reply:
I don't think we should.

The above snippet is different from the below one.  I don't  see the value in DRYing this

https://github.com/Expensify/App/blob/511a5ffc93e9afea65070e727d7df59093f10a40/src/pages/workspace/WorkspaceInvitePage.tsx#L177-L183

Reply:
What should it be called?

Reply:
How does `filterMembersBySearchTerm` sound?

Reply:
done